### ISSUE TITLE
Stolen take aim wont give Rubick the headshots

### ISSUE THREAD

**Issue by Usera2** [jeffhill/Dota2#481]
### Description

Not sure if its intended or not, but shouldn't Rubick gets that 100% headshot buff if he uses the take aim ability?

**Comment by Daar543** [comment]
Well, he already gets the range bonus from Take Aim, so the steal is "useful" for him. However, he does not have any benefit from activating that spell, so it would make sense to include it as well.

**Comment by ndakota** [comment]
I think it is by design. Aim wont give headshots for sniper ether if you never skilled headshot

**Comment by DrQuint** [comment]
> I think it is by design. Aim wont give headshots for sniper ether if you never skilled headshot

Heavily disagreed. Eclipse isn't useless if Rubick steals it off a Luna, even tho she gets no benefit either from casting it without lucent beam leveled.

**Comment by xmraz** [comment]
> I think it is by design. Aim wont give headshots for sniper ether if you never skilled headshot

Agreed. Take Aim simply changes proc chance from 40% to 100%, but if the skill is not learned (either by Rubick or Sniper himself) then there is no skill to apply on 100% of your hits.

Interestingly stolen Void's Time Walk by Rubick with Aghanim Scepter does apply Time Lock proc. Phrasing here is "applies Time Lock" instead of "100% chance to Time Lock".

**Comment by xmraz** [comment]
> > I think it is by design. Aim wont give headshots for sniper ether if you never skilled headshot
> 
> Heavily disagreed. Eclipse isn't useless if Rubick steals it off a Luna, even tho she gets no benefit either from casting it without lucent beam leveled.

Tested on 2022-06-04 (7.31c) and not working. Luna's Eclipse did 0 dmg, Rubick's Eclipse did 0 dmg (both with and without Aghanim).

**Comment by deliriam** [comment]
You need the headshot passive too right?

**Comment by DrQuint** [comment]
> Tested on 2022-06-04 (7.31c) and not working. 

You're just describing exactly what I meant. You also gave Void as another example. Rubick will get the full active benefits of a second skill when stealing a skill that depends on it. 

So by all means, for consistency, Rubick should either get the 100% headshot proc chance only while Take Aim is active (and no passive 40%), or he should have take aim be a restricted spell.

**Comment by jere535** [comment]
Rubick has had many similar inconsistencies and most of them have got a fix, no reason for heroes to have a free spell to prevent rubick from stealing actually useful stuff from them.

**Comment by Daar543** [comment]
> Rubick has had many similar inconsistencies and most of them have got a fix, no reason for heroes to have a free spell to prevent rubick from stealing actually useful stuff from them.

What is wrong with being able to steal take aim (without the headshot), or idk, tusk kick or vampiric spirit?

**Comment by Daar543** [comment]
> > I think it is by design. Aim wont give headshots for sniper ether if you never skilled headshot
> 
> Agreed. Take Aim simply changes proc chance from 40% to 100%, but if the skill is not learned (either by Rubick or Sniper himself) then there is no skill to apply on 100% of your hits.
> 
> Interestingly stolen Void's Time Walk by Rubick with Aghanim Scepter does apply Time Lock proc. Phrasing here is "applies Time Lock" instead of "100% chance to Time Lock".

Personally, I think it depends on whether the Scepter upgrade would have any effect otherwise. Since Time walk Is marked as upgradable by Scepter, there should be an upgrade with the bash. Rubick should obviously get not the bash on his normal attacks.

Example with Luna is off, as when she does not have tze Lucent Beam learned, Eclipse gives her as much as Rubick gets when he steals it.

------

There are some spells when this extra steal is not necessary. For normal spells, Charge of Darkness does not have to apply Greater Bash, and for shard upgrades, Fissure does not need to apply the aftershock effect (Rubick already got lower cooldown and free pathing from the upgrade).

**Comment by jere535** [comment]
> > Rubick has had many similar inconsistencies and most of them have got a fix, no reason for heroes to have a free spell to prevent rubick from stealing actually useful stuff from them.
> 
> What is wrong with being able to steal take aim (without the headshot), or idk, tusk kick or vampiric spirit?

Take aim is basically a passive for rubick, with a button that only slows you. Thats just a cheap instant cast ability for sniper to prevent rubick from stealing his actually useful abilities.

Tusk kick and vampiric spirit are both fine and useable on rubick.

**Comment by jere535** [comment]
> There are some spells when this extra steal is not necessary. For normal spells, Charge of Darkness does not have to apply Greater Bash, and for shard upgrades, Fissure does not need to apply the aftershock effect (Rubick already got lower cooldown and free pathing from the upgrade).

I don't agree, charge of darkness states that it applies greater bash, so obviously it should also do it for rubick.

And fissure shard upgrade states that it applies the aftershocks, so obviously it should work as well.

Ideally, every single spell rubick steals should have the same effects as the original caster, its literally the point of the hero.

**Comment by MrDemonRush** [comment]
As a rule, when Rubick steals a spell that depends on another skill he should also get the same benefit as that hero with the same skillpoints spent as that hero. This is why Requiem and Eclipse work for him.
